sdk/eas/src/types.ts

```diff
@@ -29,13 +37,225 @@ export interface SchemaField {
+  /** Reference UID (use ZERO_BYTES32 for no reference) */
+  refUID: Hex;
+  /** Encoded attestation data */
+  data: Hex;
+  /** Value sent with the attestation */
+  value: bigint;
+}
+
+/**
+ * Attestation request
+ */
+export interface AttestationRequest {
+  /** Schema UID to attest against */
+  schema: Hex;
+  /** Attestation data */
+  data: AttestationData;
+}
+
+/**
+ * Transaction result
+ */
+export interface TransactionResult {
+  /** Transaction hash */
+  hash: Hex;
+  /** Whether the transaction was successful */
+  success: boolean;
+}
+
+/**
+ * Schema information
+ */
+export interface SchemaData {
+  /** Schema UID */
+  uid: Hex;
+  /** Resolver contract address */
+  resolver: Address;
+  /** Whether attestations can be revoked */
+  revocable: boolean;
+  /** Schema string */
+  schema: string;
+}
+
+/**
+ * Attestation information
+ */
+export interface AttestationInfo {
+  /** Attestation UID */
+  uid: Hex;
+  /** Schema UID */
+  schema: Hex;
+  /** Address that created the attestation */
+  attester: Address;
+  /** Recipient of the attestation */
+  recipient: Address;
+  /** Creation timestamp */
+  time: bigint;
+  /** Expiration timestamp */
+  expirationTime: bigint;
+  /** Whether this attestation can be revoked */
+  revocable: boolean;
+  /** Reference UID */
+  refUID: Hex;
+  /** Encoded attestation data */
+  data: Hex;
+  /** Value sent with the attestation */
+  value: bigint;
+}
+
+/**
+ * Options for retrieving schemas
+ */
+export interface GetSchemasOptions {
+  /** Maximum number of schemas to return */
+  limit?: number;
+  /** Number of schemas to skip */
+  offset?: number;
+}
+
+/**
+ * Options for retrieving attestations
+ */
+export interface GetAttestationsOptions {
+  /** Maximum number of attestations to return */
+  limit?: number;
+  /** Number of attestations to skip */
+  offset?: number;
+  /** Filter by schema UID */
+  schema?: Hex;
+  /** Filter by attester address */
+  attester?: Address;
+  /** Filter by recipient address */
+  recipient?: Address;
+}
+
+/**
+ * Contract deployment result
+ */
+export interface DeploymentResult {
+  /** Deployed EAS contract address */
+  easAddress: Address;
+  /** Deployed Schema Registry contract address */
+  schemaRegistryAddress: Address;
+  /** EAS deployment transaction hash (when address not immediately available) */
+  easTransactionHash?: Hex;
+  /** Schema Registry deployment transaction hash (when address not immediately available) */
+  schemaRegistryTransactionHash?: Hex;
+}
+
+/**
+ * @deprecated Use SchemaRequest instead
+ */
+export interface RegisterSchemaOptions extends SchemaRequest {}
+
+// Internal Portal response types (not exported to users)
+
+/**
+ * @internal
+ * Portal transaction response structure
+ */
+export interface PortalTransactionResponse {
+  transactionHash?: string;
+  contractAddress?: string;
+}
+
+/**
+ * @internal
+ * Portal schema response structure
+ */
+export interface PortalSchemaResponse {
```

Comment:
Replace these with types mapped to the query

see https://gql-tada.0no.co/reference/gql-tada-api#resultof